src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,228 @@
+@doc raw"""
+    quantum_symmetric_group(n::Int)
+
+Get the Ideal that defines the quantum symmetric group on `n` elements. It is comprised of `2*n + n^2 + 2*n*n*(n-1)` many generators.
+
+# The relations are:
+
+  - row and column sum relations: `2*n` relations
+  - idempotent relations: `n^2` relations
+  - relations of type `u[i,j]*u[i,k]` and `u[j,i]*u[k,i]` for `k != j`: `2*n*n*(n-1)` relations
+
+# Output
+
+
+# Example
+
+```jldoctest
+S4 = quantum_symmetric_group(4)
+length(gens(S4))
+# output
+
+120
+```
+
+
+"""
+function quantum_symmetric_group(n::Int)
+  generator_strings = String[]
+
+  for i in 1:n, j in 1:n
+    push!(generator_strings, "u[$i,$j]")
+  end
+
+  A, g = free_associative_algebra(Oscar.QQ, generator_strings)
+  u = Matrix{elem_type(A)}(undef, n, n)
+  for i in 1:n, j in 1:n
+    u[i, j] = g[(i-1)*n + j]
+  end
+
+  relations = elem_type(A)[]
+
+  #Idempotent relations
+  for i in 1:n, j in 1:n
+    new_relation = u[i, j] * u[i, j] - u[i, j]
+    push!(relations, new_relation)
+    for k in 1:n
+      if k != j
+        new_relation = u[i,j] * u[i, k]
+        push!(relations, new_relation)
+        new_relation = u[j, i]*u[k, i]
+        push!(relations, new_relation)
+      end
+    end
+  end
+
+  #row and column sum relations
+  for i in 1:n
+    new_relation_row = -1
+    new_relation_col = -1
+    for k in 1:n
+      new_relation_row += u[i,k]
+      new_relation_col += u[k,i]
+    end
+    push!(relations, new_relation_row)
+    push!(relations, new_relation_col)
+
+  end
+  return ideal(relations)
+end
+
+@doc raw"""
+    _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+
+Get the indices of the relations that define the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+```jldoctest
+M = uniform_matroid(3,4)
+bases(M)
+idx = Oscar._quantum_automorphism_group_indices(M,:bases)
+length(idx)
+
+# output
+
+1920
+"""
+function _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+  @req structure in [:bases, :circuits, :flats] "structure must be one of :bases, :circuits, :flats"
+
+  n = length(M)
+  grdSet = matroid_groundset(M)
+
+  b    = [[] for _ in 1:n]
+  nb   = [[] for _ in 1:n]
+  rels = [[] for _ in 1:n]
+
+  sets  = eval(structure)(M)
```

Comment:
Use of `eval` is a code smell. Here it seems you want to get the Oscar function with the corresponding name, so you could do this (untested)
```suggestion
  sets  = getproperty(Oscar, structure)(M)
```

Reply:
Oh, thank you i didn't know about that function :)